src/main/kotlin/com/curiouscreature/kotlin/math/Matrix.kt

```diff
@@ -386,7 +386,78 @@ fun transpose(m: Mat4) = Mat4(
         Float4(m.x.w, m.y.w, m.z.w, m.w.w)
 )
 fun inverse(m: Mat4): Mat4 {
-    TODO("Implement inverse(Mat4)") // TODO
+    val result = Mat4()
+    val pairs = FloatArray(12)
+
+    pairs[0]  = m.z.z * m.w.w
+    pairs[1]  = m.w.z * m.z.w
+    pairs[2]  = m.y.z * m.w.w
+    pairs[3]  = m.w.z * m.y.w
+    pairs[4]  = m.y.z * m.z.w
+    pairs[5]  = m.z.z * m.y.w
+    pairs[6]  = m.x.z * m.w.w
+    pairs[7]  = m.w.z * m.x.w
+    pairs[8]  = m.x.z * m.z.w
+    pairs[9]  = m.z.z * m.x.w
+    pairs[10] = m.x.z * m.y.w
+    pairs[11] = m.y.z * m.x.w
+
+    result.x.x = pairs[0] * m.y.y + pairs[3] * m.z.y + pairs[4] * m.w.y
```

Comment:
Could you check the code generated by the Kotlin compiler to make sure these assignments are efficient?

Reply:
Here is generated code
https://pastebin.com/uKjbtyiR 
Everything looks fine

Reply:
Thanks. Hopefully the VM will get rid of all those `invokevirtual`s.